### ISSUE TITLE
Nuc not waking up properly

### ISSUE THREAD

**Issue by Tobix99** [zearp/Nucintosh#103]
Hey there,

I've upgraded my EFI and MacOS a while ago (in October) to MacOS Ventura 13.0.1. Since then I've the Bug, that my Nuc would start to sleep, but not wake up properly. It will not respond to pressing any Keys and the Power button. Only holding the Power button and resetting/rebooting is the only way to get back in. The whole Time the blue Power-Led is lit up.

The issue occurs not at every sleep, I don't know jet what the trigger is.

I've installed a Broadcom Wifi/Bluetooth Card.

My ```pmset``` options are:

```
>>> pmset -g

System-wide power settings:
Currently in use:
 standby              1
 Sleep On Power Button 1
 womp                 0
 autorestart          0
 hibernatefile        /var/vm/sleepimage
 proximitywake        1
 powernap             0
 gpuswitch            2
 networkoversleep     0
 disksleep            10
 standbydelayhigh     86400
 sleep                1 (sleep prevented by sharingd)
 hibernatemode        3
 ttyskeepawake        1
 displaysleep         30
 tcpkeepalive         1
 highstandbythreshold 50
 standbydelaylow      86400
```

I'm using the sleep extensively, because the boot-time at 4 minutes is a bit long to boot it every day (imo).

I've read in the Readme, that ```standby``` and ```autopoweroff``` have to be zero, is this the problem? I've never changed these values, did the Upgrade change this?

**Comment by Tobix99** [comment]
I've now set ```standby```, ```autopoweroff```, ```proximitywake``` and ```hibernatemode``` to 0.

And will report back if it's still crashing

**Comment by zearp** [comment]
Yes, those might be related to it not waking up properly. I've updated my main NUC to Ventura 13.0.1 which also has Broadcom and have no issues with waking up from sleep using the settings from the readme. If your keyboard and mouse are connected via bluetooth you'll have to user the power button to wake up the machine.

Surprisingly on my test NUC this stock I can still wake up the machine using Intel bluetooth. That has never worked on previous macOS versions. Broadcom is unable to do that without modifying the hardware itself as the power is cut to the bluetooth module during sleep.

**Comment by zearp** [comment]
Yes, those might be related to it not waking up properly. I've updated my main NUC to Ventura 13.0.1 which also has Broadcom and have no issues with waking up from sleep using the settings from the readme. If your keyboard and mouse are connected via bluetooth you'll have to use the power button to wake up the machine.

Surprisingly on my test NUC this stock I can still wake up the machine using Intel bluetooth. That has never worked on previous macOS versions. Broadcom is unable to do that without modifying the hardware itself as the power is cut to the bluetooth module during sleep.

**Comment by Tobix99** [comment]
Hey thanks for the Info, but they are connected via USB, I don’t have a fancy Bluetooth Keybaord 👀

But since yesterday it seems to be working… I will report back in like 1-2 weeks, because it was a sporadic problem.

But I think a OpenCore or MacOS Update must have changed the settings, because I didn’t….weird.

**Comment by zearp** [comment]
Yes, system updates and upgrades can reset some of the power management settings. It doesn't always happens but it can which is bad, updates shouldn't change user changes setting to something else, specially not without notification but Apple also turns on telemetry when updating sometimes, which I personally believe is even worse. More than once an updated enabled some of these settings despite me making na effort to always make sure they are turned off.

![Screenshot 2022-11-25 at 18 57 06](https://user-images.githubusercontent.com/63272687/204036331-d75e90ad-3885-453a-a9a5-86ce8264961c.png)

You can quickly check and fix it with Hacking tool if you update the system again. Any "bad" setting will show up as red and clicking on the screwdriver fixes it. You can also re-run the commands from the readme to fix them.

![Screenshot 2022-11-25 at 18 57 53](https://user-images.githubusercontent.com/63272687/204036403-8ee8b4e3-0fbf-4e3b-8cd6-2825238862a9.png)

I noticed some of the things I had turned off show as 1 (enabled) but nothing that would mess with sleep. I just like to keep things such as tcp keep alive and others. I always turn those off because it causes (semi) wake-ups by design. It can half wake up the machine to check email or make a backup. I don't want any of that, yet it gets enabled form time time when updating. It's part of the macOS experience I guess lol.

Good luck!

**Comment by zearp** [comment]
My boot times are under a minute but it differs per machine. I will see if I can ave one sleep for a few days to see if I can reproduce the issue. If I can't reproduce it there's not much I can do unfortunately.

What I find odd is that you're never greeted with a crash report on the next after you had to force it to power off. This either mean there is no panic, or it isn't logged or some settings in the config have been changed (unlikely if you followed the readme).

So really no clue what I can do other than try to let a NUC sleep for a few days and see if it turns on or not.

**Comment by Tobix99** [comment]
Thank you! But if not don't worry...I think I also found the device responsible for my long boot times: My Samsung SATA SSD.

[Reddit](https://www.reddit.com/r/hackintosh/comments/sfqhcc/comment/i5m8u5t/?utm_source=share&utm_medium=web2x&context=3)

With shorter boot times I think I don't need the sleep that often anymore, but it still bugs me what it could be.

After it turn "on"  (in the half wake state) from the sleep nothing else turn "on". My USB Hub has some LEDs to see if a USB Device is connected. These are off when it's sleeping and are not turning on like normal when it's half awake. The Monitor is also not waking up.

When the NUC is properly waking up (like pressing keys on the keyboard) everything turn on normally.

When it is in this half wake state nothing is registered, no power button press (except the really long on like 10 seconds) no keyboard buttons. I can only cut the power or press the button long to reset the NUC. After that it boots like it was restated. It even opens all previous windows. No crash reports or panics.

**Comment by zearp** [comment]
Which model do you have? I have no issues on my 860 EVO. But thats the only Samsung SSD I own.

**Comment by Tobix99** [comment]
Samsung SSD 850 EVO 1TB

"Double Booting" with Refind and Opencore, because it has 3 Partitions --> Ubuntu, MacOSX and Windows (but rarely use them). But that shouldn't have any impact.

**Comment by zearp** [comment]
Well the bios is more tricky to answer, I wouldn't update if I were you. After updating to 0090 my under voting stopped working and even force downgrading to 0083 didn't restore it. Intel has disabled it somewhere somehow without na option to undo it. It being the holidays now I have too little time to try and crack the nut now. So I would advise not updating the bios. The chance your issue is caused by the bios is very small. I have been running macOS on this since bios versions in the 0070 range.

**Comment by gsternagl** [comment]
Folks,

I have a similar issue as Tobix99. Main differences here are:

1. Wake-up via keyboard or mouse (both USB) work fine
2. Wake-up via scheduled alarm wake the machine up but then it freezes without a kernel panic and I can only reboot it then. As these scheduled alarms happen frequently, I cannot use sleep at anymore.

I had no issue with exactly this same machine with Big Sur or Monterey. It all started when I updated to Ventura. And yes I have your EFI and the latest OpenCore version.  I was able to trace it down to be related to Thunderbolt somehow. I have a second display connected via Thunderbolt / Displayport. When I unconnect that display, software wakeup works fine again. When I put it back and reboot (no hotplug enabled), software wakeups freeze the machine again. I have Thunderbold security in legacy mode and TB-firmware is unpatched.

What I tried:
- run OSX in safe-mode (to avoid loading 3rd party drivers) => issue is still there
- rebuild the kext-cache or whatever it's called these days with Onyx => issue persists
- booted in text-mode to see whether there are any obscure messages => nothing out of the normal
- updated the NUC8 BIOS to the latest, went back to stock settings and then re-enabled all settings again.
- I have exactly the same NUC8i5BEK with the same HW, BIOS-settings, PM-settings, Coreboot and OSX versions but no Thunderbolt => there I don't have this wakeup issues.


Originally the "faulty" NUC8 was upgraded from Monterey to Ventura but I had some issues with it. So I re-installed and then restored my last Timemachine backup. As this issue with Thunderbolt didn't exist up to Monterey, I still suspect that there are some left-overs from my previous Monterey or Big Sur days, brought in through Timemachine restore. I will try to install a fresh Ventura installation on an external SSD and see whether the issues persist there too.

**Comment by Tobix99** [comment]
For anyone interested: I fixed it!

I'm currently running MacOS 13.7.1 with the latest EFI (Sequoia only EFI (05102024)).

I'm not using the onboard Intel WIFI/Bluetooth (but a Broadcom M.2 module instead).

Following steps to fix the "sleep problem"
- disabling the following KEXTs: `IntelBluetoothFirmware.kext, BlueToolFixup.kext, NVMeFix.kext and IntelBTPatcher.kext`
- Running the Power Commands again

```bash
sudo pmset standby 0
sudo pmset autopoweroff 0 
sudo pmset proximitywake 0
sudo pmset powernap 0 
sudo pmset tcpkeepalive 0
sudo pmset womp 0
sudo pmset hibernatemode 0
```

Now the sleep is working again as expected.

**Comment by zearp** [comment]
Oh, maybe I need to clarify that in the readme but if you're using Broadcom (on Ventura cuz newer I don't support) make sure in the BIOOS the onboard wifi and bluetooth are disabled plus remove any related kext to Intel wifi or bluetooth, removing the NVMe driver is also fine since it's not using an m2 ssd.

As for newer versions go, the yuwill work but you have to patty the file system which will reduce security, which is easy to deal with but you'll also no longer run vanilla macOs which iOS one of the goals of my EFI. So I can't really support it but it will work just fine with OpenCore Legacy Patcher root patches applied.

tl;dr
Broadcom users: disable any Intel kext, ?NVMe kext and also make sure the wifi and bluetooth are disabled in the bios.

**Comment by zearp** [comment]
Oh, maybe I need to clarify that in the readme but if you're using Broadcom (on Ventura cuz newer I don't support) make sure in the bios the onboard wifi and bluetooth are disabled plus remove any related kext to Intel wifi or bluetooth, removing the NVMe driver is also fine since it's not using an m2 ssd.

As for newer versions of macOS go, they will work but you have to patch the file system which will reduce security, which is easy to deal with but you'll also no longer run vanilla macOS which is one of the goals of my EFI. So I can't really support it but it will work just fine with OpenCore Legacy Patcher root patches applied.

tl;dr
Broadcom users: disable any Intel kext, NVMe kext and also make sure the wifi and bluetooth are disabled in the bios.

**Comment by gsternagl** [comment]
Thanks for the update. My problems went away as well but I have taken a different route but probably with the same underlying change.
First off all I need both NVMeFix.kext and BlueToolFixup.kext. So they were always installed on my system and still are. I don't have the Intel bluetooth stack as I am using an ASUS USB Bluetooth stick. But that is not really so great as it doesn't work great with devices that produce a lot of data like e.g. high resolution headphones. I've given up on that and bought a little conference system from Jabra instead which is connected via USB.
But the approach that I have taken to get sleep working again was to revert back and just start with a fresh install of Zearp's original OpenCore configuration and then just do a couple of minor tweaks to customize my boot menu. Since then sleep has been working fine. I am now running Sonoma. But my installation has been stabile and has also been upgraded multiple times with all major MacOS versions since 2021. I am using this system on daily basis and it's been rock solid. Big shout out to Zearp!
